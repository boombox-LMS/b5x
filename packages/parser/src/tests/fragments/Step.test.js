const {
  exampleStepMarkup,
} = require("../../../dist/componentBuilders/fragments/parsingClasses/Step");
const { RawFragmentSchema } = require("@b5x/types");

describe("The Step fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleStepMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed step fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed step fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});
