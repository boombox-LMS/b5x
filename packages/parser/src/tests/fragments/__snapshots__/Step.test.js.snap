// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Step fragment builds as expected The parsed step fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [],
  "contentType": "Step",
  "contents": "",
  "data": Object {
    "title": "Initialize your topic",
  },
  "dependencyUris": Array [],
  "digest": "e902a3674b44714da75ade97b77c4167",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;
